[Autograd Testing] Add a test where child reentrant task fails.

pritamdamania87 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#35223 [Autograd Testing] Add a test where child reentrant task fails.**
finishes.
* **#35223 [Autograd Testing] Add a test where child reentrant task fails.**

Adding tests as part of
https://github.com/pytorch/pytorch/issues/34367.

This test covers
"Mixed with errors" ->
"Reentrant on same device" ->
"Make child error before parent finishes"

Differential Revision: [D20603127](https://our.internmc.facebook.com/intern/diff/D20603127/)

pritamdamania committed:
[Autograd Testing] Add a test where child reentrant task fails.

Adding tests as part of
https://github.com/pytorch/pytorch/issues/34367.

This test covers
"Mixed with errors" ->
"Reentrant on same device" ->
"Make child error before parent finishes"

Differential Revision: [D20603127](https://our.internmc.facebook.com/intern/diff/D20603127/)

[ghstack-poisoned]

wanchaol reviewed (approved):
looks good.

albanD commented on test/test_autograd.py:
```diff
@@ -2142,6 +2142,42 @@ def backward(ctx, grad_output):
+
+        # Parent graph.
+        a = torch.rand(3, 3, requires_grad=True)
+        b = torch.rand(3, 3, requires_grad=True)
```
nit: maybe declare `b` below to make it obvious that the two graphs are independent.

albanD reviewed (approved):
LGTM

gqchen reviewed (approved):
looks good to me.

pritamdamania committed:
Update on "[Autograd Testing] Add a test where child reentrant task fails."

finishes.
* **#35223 [Autograd Testing] Add a test where child reentrant task fails.**

Adding tests as part of
https://github.com/pytorch/pytorch/issues/34367.

This test covers
"Mixed with errors" ->
"Reentrant on same device" ->
"Make child error before parent finishes"

Differential Revision: [D20603127](https://our.internmc.facebook.com/intern/diff/D20603127/)

[ghstack-poisoned]
